sda-commons-server-kafka/src/test/java/com/salesforce/kafka/test/SharedKafkaTestResourceScram.java

```diff
@@ -0,0 +1,100 @@
+package com.salesforce.kafka.test;
+
+import com.salesforce.kafka.test.junit4.SharedKafkaTestResource;
+import com.salesforce.kafka.test.junit4.SharedZookeeperTestResource;
+import com.salesforce.kafka.test.listeners.BrokerListener;
+import com.salesforce.kafka.test.listeners.SaslScramListener;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import kafka.admin.ConfigCommand;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * Creates and starts up a {@link SharedKafkaTestResource} that uses a custom {@link
+ * SharedZookeeperTestResource} with installed SCRAM-SHA-* credentials.
+ *
+ * <p>See {@link SharedKafkaTestResource} for usage.
+ */
+public class SharedKafkaTestResourceScram extends SharedKafkaTestResource {
+  private final SharedZookeeperTestResource sharedZookeeperTestResource =
+      new SharedZookeeperTestResource();
+
+  @Override
+  protected void setKafkaCluster(KafkaCluster kafkaCluster) {
+    // we need to replace the zkTestServer in the kafka cluster in order to change the Zookeper
+    // configuration after it started and before the kafka broker starts.
+    if (kafkaCluster instanceof KafkaTestCluster) {
+      try {
+        Field f = KafkaTestCluster.class.getDeclaredField("zkTestServer");
+
+        // the field is private, so we need to make it accessible
+        f.setAccessible(true);
+
+        // the field is final, so we need to remove the modifier
+        Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(f, f.getModifiers() & ~Modifier.FINAL);
```

Comment:
Do we have a test helper class for these kind of modifications? I imagine we need this trick in a couple of places.

Reply:
I'm not sure. I guess we shouldn't make it too easy to do these kinds of things.